common/source/serial_port.cpp

```diff
@@ -170,9 +194,23 @@ error_condition SerialPort::Get(const string& key, string& value) {
 	return kNoError;
 }
 
-/*
- * @brief SerialPort support half-duplex communication
- */
+void read_callback(bool& data_available, deadline_timer& timeout, \
```

Comment:
Please try not to put functions outside a class if it is related to some classes.